script.js

```diff
@@ -1,4 +1,238 @@
+      generatePlayerScore(input);
+      player1Score = Number(playerScore);
+      // Add score to accumulate total score
+      totalp1Score += player1Score;
+      myOutputValue = 'Player 1, your score is: ' + player1Score + ". Player 2's turn. Click submit to roll.";
+      gameMode = 'Player 2 Variable Dice Mode';
+    }
+    else {
+      myOutputValue = invalidMessage;
+    }
+    // Empty the global variable of helper function output to create a new value for player 2.
+    diceValues = [];
+    dicePosition = [];
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 2 Variable Dice Mode') {
+    // List the dice index using p1DiceValues with the helper function
+    // Match it to Player 2
+    getDicePosition(p1DiceValues.length);
+    p2DicePosition = dicePosition;
+    // List the dice values using p1DiceValues with the helper function
+    // Match it to Player 2
+    getNumOfDiceAndRoll(p1DiceValues.length);
+    p2DiceValues = diceValues;
+    gameMode = 'Player 2 Enter Position';
+    // Output Player 2 dice values and prompt index question.
+    myOutputValue = 'Player 2, you have decided to <br>play with ' + numOfDice + ' dice. <br><br>You rolled: <br>---- dice: ' + p2DiceValues + '<br>position: ' + p2DicePosition + '<br><br>Enter the positions of the numbers you want.';
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 2 Enter Position') {
+    // Check if input is a number and is not an empty string.
+    if (isNaN(Number(input)) == false && input != '' && input.length == p2DiceValues.length) {
+      // Put input into helper function to generate player's score
+      generatePlayerScore(input);
+      player2Score = Number(playerScore);
+      totalp2Score += player2Score;
+      myOutputValue = 'Player 2, your score is: ' + player2Score;
+      // On click, auto compute the total scores to determine the winner.
+      gameMode = 'Compare Scores';
+    }
+    else {
+      myOutputValue = invalidMessage;
+    }
+    // Empty the global variable of helper function output to create a new value for player 1.
+    diceValues = [];
+    dicePosition = [];
+    return myOutputValue;
+  }
+  // Determine the winner by comparing scores.
+  if (gameMode == 'Compare Scores') {
+    if (totalp1Score > totalp2Score) {
+      myOutputValue = 'Player 1 score: ' + totalp1Score + '<br>Player 2 score: ' + totalp2Score + '. <br><br>Player 1 wins! Key in "normal" or "variable" to restart game.';
+    }
+    else if (totalp1Score < totalp2Score) {
+      myOutputValue = 'Player 1 score: ' + totalp1Score + '<br>Player 2 score: ' + totalp2Score + '. <br><br>Player 2 wins! Key in "normal" or "variable" to restart game.';
+    }
+    else {
+      myOutputValue = 'It is a draw! Key in "normal" or "variable" to restart game.';
+    }
+    // Change game mode to the start before returning it.
+    gameMode = 'Enter Mode';
+    return myOutputValue;
+  }
+
+  // Start Normal Dice Game
```

Comment:
Nice- relatively straightforward logic for this section. I would recommend abstracting each "version" of the game into a helper function to keep `main` organised and decluttered.